[feat] Dependencies Explorer

dobrishinov opened the pull request:
## Describe your changes

Dependencies Explorer
=====================

Interactive UI and model for analyzing Salesforce metadata dependencies, visualizing relationships, and exporting package.xml files or human-readable reports.

<img width="1909" height="372" alt="image" src="https://github.com/user-attachments/assets/e58bf254-276c-423a-b74d-d8c4b0cc481d" />

<img width="1916" height="543" alt="image" src="https://github.com/user-attachments/assets/8df13635-22a8-4a31-a685-43452d63d88c" />

<img width="1920" height="507" alt="image" src="https://github.com/user-attachments/assets/5e16f8a4-0a2b-4f7e-b7fb-4880152db25d" />

<img width="1920" height="904" alt="image" src="https://github.com/user-attachments/assets/a74c8722-0b2b-40ce-9acf-4032d410420f" />

<img width="1920" height="950" alt="image" src="https://github.com/user-attachments/assets/ed3e2cf4-273a-449f-a849-93ea4daa5c0b" />


What It Is
----------

Dependencies Explorer uses the Salesforce Tooling API to explore metadata relationships in both directions ("Depends On" for what a component references, and "Referenced By" for what references it). It provides a searchable UI for selecting items, displays results in quick summaries or dependency trees, and supports exports for packaging or review.

Why Use It
----------

*   Accelerates impact analysis, troubleshooting, and change planning by revealing hidden dependencies (e.g., lookups, value sets, dependent picklists).
    
*   Generates minimal package.xml files for efficient metadata retrieval.
    
*   Offers readable reports for team reviews, beyond what's visible in file names or standard tools.
    

Key Capabilities
----------------

*   **Metadata Selection**: Searchable dropdown for types and items; excludes managed packages by default (with toggle).
    
*   **Analysis Directions**: "Depends On" (references) and "Referenced By" (referencers).
    
*   **Views**:
    
    *   Quick Summary: Deduplicated list grouped by type, ideal for exports.
        
    *   Dependency Tree: Nested parent-child structure for interactive exploration.
        
*   **Recursive Resolution**: Fetches dependencies via Tooling API (MetadataComponentDependency), with special handling for types like CustomField, Flow, and ValidationRule.
    
*   **Enrichments**: Maps custom fields to objects, resolves lookups, global value sets, and dependent picklists; synthesizes missing dependencies for unsupported metadata.
    
*   **Exports**:
    
    *   Creates a package.xml from the current dependency set.
        
    *   Generates a human-readable report with stats, per-type listings, and flat summary format.
        
*   **Interactivity**: Click-through links to open items in Salesforce (where IDs are available); JSON debug view for raw data inspection.
    
*   **Performance Guards**: Detects circular references; supports batching or Composite API for large graphs.
    

How It Works (High-Level)
-------------------------

*   Queries Tooling API to recursively gather dependency records.
    
*   Enhances data by querying CustomField and CustomObject metadata for readable names and supplemental relationships.
    
*   Builds in-memory views (grouped or nested) and renders them.
    

Limitations
-----------

*   Relies on Tooling API.
    
*   Synthetic dependencies (e.g., value sets, controlling picklists) lack IDs and can't be opened in Salesforce.
    

Quick Usage
-----------

1.  Select a metadata type and item via the searchable dropdown (toggle to include/exclude managed packages) .
    
2.  Choose analysis direction ("Depends On" or "Referenced By").
    
3.  Toggle between Quick Summary and Dependency Tree views.
    
4.  Export package.xml or a readable summary report once results load.

## Issue ticket number and link
https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/963

## Checklist before requesting a review
- [x] I have read and understand the [Contributions section](https://github.com/tprouvot/Salesforce-Inspector-reloaded#contributions)
- [x] Target branch is releaseCandidate and not master
- [x] I have performed a self-review of my code
- [x] I ran the [unit tests](https://github.com/tprouvot/Salesforce-Inspector-reloaded#unit-tests) and my PR does not break any tests
- [ ] I documented the changes I've made on the [CHANGES.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/CHANGES.md) and followed actual conventions
- [ ] I added a new section on [how-to.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/docs/how-to.md) (optional)

CamilleGuillory committed:
[feat] Flow Scanner (#889)

## Describe your changes

## Issue ticket number and link

## Checklist before requesting a review
- [x] I have read and understand the [Contributions
section](https://github.com/tprouvot/Salesforce-Inspector-reloaded#contributions)
- [x] Target branch is releaseCandidate and not master
- [x] I have performed a self-review of my code
- [ ] I ran the [unit
tests](https://github.com/tprouvot/Salesforce-Inspector-reloaded#unit-tests)
and my PR does not break any tests
- [ ] I documented the changes I've made on the
[CHANGES.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/CHANGES.md)
and followed actual conventions
- [ ] I added a new section on
[how-to.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/docs/how-to.md)
(optional)

---------

Co-authored-by: Camille.GUILLORY <Camille.GUILLORY@adeccogroup.com>
Co-authored-by: Thomas Prouvot <prouvot.t@gmail.com>

tprouvot committed:
Enable custom banner text on sandboxes

tprouvot committed:
Remove hardcoded api version

tprouvot committed:
Update doc to be able to render mermaid js schema

tprouvot committed:
Merge branch 'releaseCandidate' of https://github.com/tprouvot/Chrome-Salesforce-inspector into releaseCandidate

tprouvot committed:
Update flow scanner doc

DivyanshuBist committed:
[feat] Field Creator:  Field Creation Permission CheckBox UI enhancement (#936)

## Describe your changes
this PR fixes [Issue
931](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/931),
we have added following conditions on checkbox Event Handler:-
1) if edit is **check**, **check** read as well
2) if read is **uncheck**, **uncheck** edit as well

## Issue ticket number and link
[Issue
931](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/931)

## Checklist before requesting a review
- [x] I have read and understand the [Contributions
section](https://github.com/tprouvot/Salesforce-Inspector-reloaded#contributions)
- [x] Target branch is releaseCandidate and not master
- [x] I have performed a self-review of my code
- [ ] I ran the [unit
tests](https://github.com/tprouvot/Salesforce-Inspector-reloaded#unit-tests)
and my PR does not break any tests
- [x] I documented the changes I've made on the
[CHANGES.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/CHANGES.md)
and followed actual conventions
- [ ] I added a new section on
[how-to.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/docs/how-to.md)
(optional)

Co-authored-by: Thomas Prouvot <prouvot.t@gmail.com>

tprouvot committed:
Remove beta tag for Field Creator

tprouvot committed:
Merge branch 'releaseCandidate' of https://github.com/tprouvot/Chrome-Salesforce-inspector into releaseCandidate

KamilGadawski committed:
[feat] Field Creator: Improve UI for Option modal (#928)

## Describe your changes
Improvement of the modal interface for Options in the Field Creator.
What has been changed:
- Uniform appearance of the modal header for Options and Permissions 
- Increased margin between labels and text area
- Vertical centering of the checkbox section

<img width="509" height="518" alt="Options_Modal"
src="https://github.com/user-attachments/assets/0e1b70ae-6339-4312-abb6-1ea83622c88b"
/>


## Issue ticket number and link

## Checklist before requesting a review
- [x] I have read and understand the [Contributions
section](https://github.com/tprouvot/Salesforce-Inspector-reloaded#contributions)
- [x] Target branch is releaseCandidate and not master
- [x] I have performed a self-review of my code
- [ ] I ran the [unit
tests](https://github.com/tprouvot/Salesforce-Inspector-reloaded#unit-tests)
and my PR does not break any tests
- [x] I documented the changes I've made on the
[CHANGES.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/CHANGES.md)
and followed actual conventions
- [ ] I added a new section on
[how-to.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/docs/how-to.md)
(optional)

---------

Co-authored-by: Kamil Gadawski <kamil.gadawski@aldi-techhub.pl>
Co-authored-by: Thomas Prouvot <prouvot.t@gmail.com>

dobrishinov committed:
Dependencies Explorer Initial Commit (stable version 1.0.0)

dobrishinov committed:
Merge branch 'releaseCandidate' into feature/Dependencies-Explorer

dobrishinov committed:
Added missing fieldCreatorBtn css class to fix the next button for the Dependencies Explorer

tprouvot committed:
Merge branch 'releaseCandidate' into pr/dobrishinov/964

tprouvot committed:
linter update

tprouvot committed:
Implement SLDS2 theme basics

tprouvot commented:
Hi @dobrishinov,
Thanks for your PR ! 
That would be a great addition for Reloaded, I have to deep dive into the code now I have implemented SLDS v2 on it.

tprouvot committed:
Merge branch 'releaseCandidate' into pr/dobrishinov/964

tprouvot committed:
Add new page available for custom shortcut configuration

tprouvot committed:
Update Dependencies Explorer button label and add keyboard shortcut

tprouvot committed:
Remove option to hide button since it is not used in popup.js

tprouvot commented on addon/dependencies-explorer.js:
```diff
@@ -0,0 +1,2766 @@
+    {value: "WebLink", label: "Custom Buttons (WebLink)"},
+    {value: "Layout", label: "Page Layouts"},
+    {value: "FlexiPage", label: "Lightning Pages"},
+    {value: "GlobalPicklist", label: "Global Picklists"}
```
How is built this list ? Do we really need to hardcode this list ? Is it available through describeMetadata() method ?

dobrishinov replied:
METADATA_TYPES is hardcoded because each supported type needs a dedicated Tooling API SOQL query and sometimes custom handling. Many results from describeMetadata() can’t be queried via Tooling API, and some have naming mismatches (e.g. LightningComponent vs AuraDefinitionBundle). Keeping this list explicit ensures reliability, avoids extra startup API calls, and maintains consistent UX and query behavior.

tprouvot commented on addon/dependencies-explorer.js:
```diff
@@ -0,0 +1,2766 @@
+
+  /** @type {Array<string>} List of standard Salesforce objects */
+  STANDARD_OBJECTS: [
+    "Account", "Contact", "Case", "Opportunity", "Lead", "User", "Profile", "Role", "Task", "Event", "Note", "Attachment",
```
How is built this list ?

dobrishinov replied:
This code is obsoleted. I missed to removed it after second refactoring iteration. Now it's removed.

tprouvot commented on addon/dependencies-explorer.js:
```diff
@@ -0,0 +1,2766 @@
+  getTypeIcon(type) {
+    const iconMap = {
+      // Code-related icons - SLDS patterns
+      "ApexClass": "m163 122-23-19c-7-5-15-4-21 2L3 251c-4 5-4 13 0 19l117 145c5 6 14 8 21 2l23-19c7-5 8-15 2-21L72 260l95-117c4-6 3-15-4-21zm354 129L400 106c-5-6-14-8-21-2l-23 19c-7 5-8 15-2 21l95 117-95 117c-5 6-4 16 2 21l23 19c7 5 15 4 21-2l117-146c4-7 4-14 0-19zM316 108l-30-7c-8-2-17 3-19 11l-74 284c-2 8 3 16 11 18l30 7c8 2 17-3 19-11l74-284c2-9-3-16-11-18z",
```
I would rather user symbols.svg for icons

dobrishinov replied:
Added this to my TODO list. I’ll try moving those SVG icons into symbols.svg, though some additional adjustments may be needed because of the custom CSS applied to them.

dobrishinov replied:
✔All metadata icons have been moved to symbols.svg. The SVGs in that file are on a single line, which is why Git shows a large number of changes.

tprouvot commented on addon/dependencies-explorer.js:
```diff
@@ -0,0 +1,2766 @@
+   * @param {string} id - The ID to validate
+   * @returns {boolean} True if valid Salesforce ID
+   */
+  isValidSalesforceId(id) {
```
You can export data-load.js isRecordId method in utils.js and reuse it

dobrishinov replied:
Done 😊 I added an isRecordId method to utils.js and updated dependencies-explorer.js. I didn’t update data-load.js to reuse it from utils.js since I didn’t want to introduce that change in this branch.

tprouvot commented on addon/dependencies-explorer.js:
```diff
@@ -0,0 +1,2766 @@
+
+  /**
+   * Core logic: recursively fetch dependencies and build a tree.
+   * This is a simplified port of sfdc-soup-master logic, using sfConn.rest for SOQL queries.
```
If some code is re-used from another lib, it must be mentioned somewhere.
Is it an inspiration or the exact same code ?

dobrishinov replied:
It's not a reused code. I checked that project for inspiration but build dependencies tree in more simply way. Now added as a comment the link to the project also 😊

tprouvot commented on addon/dependencies-explorer.js:
```diff
@@ -0,0 +1,2766 @@
+      const soql = this._buildMetadataQuery(metadataType);
+      if (!soql) return [];
+
+      let res = await sfConn.rest(`/services/data/v${apiVersion}/tooling/query/?q=` + encodeURIComponent(soql));
```
You are not checking for nextRecordsUrl response property, which means that you are missing some metadata over 2k rows returned

dobrishinov replied:
Good catch! 
I fixed it and add one new method that used nextRecordsUrl.
I don't have more then 2k results but you can check now and test if everything is fine 😊

tprouvot commented on addon/dependencies-explorer.js:
```diff
@@ -0,0 +1,2766 @@
+   * Generates a valid package.xml file from the current dependencies
+   * @returns {string} The package.xml content
+   */
+  generatePackageXml() {
```
To avoid code duplication, you can re-use generatePackageXml from metadata-retrieve.js

dobrishinov replied:
generatePackageXml needs some tweaking. Since it was added at the end, I may not have time to work on it now.

tprouvot commented on addon/dependencies-explorer.js:
```diff
@@ -0,0 +1,2766 @@
+  /**
+   * Downloads the generated package.xml file
+   */
+  downloadPackageXml() {
```
same re-use for downloadXml from metadata-retrieve.js

dobrishinov replied:
Same as generatePackageXml.

tprouvot commented on addon/dependencies-explorer.js:
```diff
@@ -0,0 +1,2766 @@
+      className: "dep-icon-blue-margin"
+    },
+    h("path", {
+      d: "m272 417-21-3-21-6c-4-1-9 0-12 3l-5 5a79 79 0 0 1-106 6 77 77 0 0 1-4-112l76-76c10-10 22-16 34-20a79 79 0 0 1 74 20l10 13c4 7 13 8 18 2l28-28c4-4 4-10 1-15l-14-16a128 128 0 0 0-71-37 143 143 0 0 0-124 37l-73 73a139 139 0 0 0-6 193 137 137 0 0 0 198 4l25-25c7-5 2-17-7-18zM456 58a139 139 0 0 0-193 6l-23 22c-7 7-2 19 7 20 14 1 28 4 42 8 4 1 9 0 12-3l5-5a79 79 0 0 1 106-6 77 77 0 0 1 4 112l ...
```
You can re-use symbols.svg#link

dobrishinov replied:
Done.

tprouvot commented on addon/dependencies-explorer.js:
```diff
@@ -0,0 +1,2766 @@
+                className: "dep-icon-inline-no-margin"
+              },
+              h("path", {
+                d: "M496 453 362 320a189 189 0 1 0-340-92 190 190 0 0 0 298 135l133 133a14 14 0 0 0 21 0l21-21a17 17 0 0 0 1-22ZM210 338a129 129 0 1 1 130-130 129 129 0 0 1-130 130Z"
```
Why is the magnify icon for ?

dobrishinov replied:
The icon is this one on the bottom left side. It's just a pure UI thing in front of the result.

Image 1
<img width="555" height="209" alt="image" src="https://github.com/user-attachments/assets/e22e6a4c-d640-4248-9a42-177db6bbc781" />

Image 2
<img width="139" height="65" alt="image" src="https://github.com/user-attachments/assets/ee8d2a97-7650-44f1-ba30-406101978e99" />

tprouvot replied:
Ok thanks, I think it brings confusion, could you remove it ?

dobrishinov replied:
Done. Removed it.

tprouvot reviewed (commented):
Hi @dobrishinov 
Thanks for your PR ! 

I have few questions / remarks.

Since your PR creation, I updated the contributions section from README, and added a general recommandation for design: the usage of SLDS for containers, principal components etc. Could you perform some changes in order to enforce its usage ?

Thanks a lot !

dobrishinov committed:
Improving fetching of the records from Tooling API, some minor tweaks and refactoring of unused code.

dobrishinov committed:
Added isRecordId to utils.js and reuse the method in the dependencies-explorer.js

dobrishinov committed:
Removed "magnify icon" from the dep-footer

dobrishinov committed:
Moved metadata icons in symbols.svg, added small css fixes

dobrishinov committed:
Polish UI for improved SLDS 2 compatibility and clarity

dobrishinov commented:
Hey, thank you for the feedback. I've updated the design to be more SLDS2 compliant, and I think it now looks more consistent and informative for users. Attached images below. 😊

<img width="1309" height="546" alt="dependency-explorer-1" src="https://github.com/user-attachments/assets/40f13859-f2fa-40b0-a221-9d5d5001c981" />
<img width="1312" height="628" alt="dependency-explorer-2" src="https://github.com/user-attachments/assets/193a402c-fd3b-4838-998a-c2e7428a9ec8" />
<img width="1312" height="692" alt="dependency-explorer-3" src="https://github.com/user-attachments/assets/7c02f0aa-6916-49d2-ac61-2507522a309e" />
<img width="1470" height="962" alt="dependency-explorer-4" src="https://github.com/user-attachments/assets/a5a643e5-2111-4fb0-84b1-557c94c220d0" />
<img width="1492" height="962" alt="dependency-explorer-5" src="https://github.com/user-attachments/assets/93341df6-ac0f-47fd-9fb9-a33f90b83967" />

tprouvot commented:
Hi @dobrishinov 
Thanks for the updates you've made ! 

I would like to address few details about the UI to make it more consistent with the other pages of the extension.

- Switch positions of the header controls, metadata selection on the left and actions buttons on the right
- Adding JSON export button with the other actions buttons relocated on the right of the screen
- main container may not be needed, it shows a difference with the other that uses border radius. You can inspire from  event-monitor page, I would like to keep a similar spirit for this page

<img width="1437" height="828" alt="image" src="https://github.com/user-attachments/assets/a8f2abb1-e224-49d9-b5df-483cc3449c5f" />

Thanks for your help on those changes, if you have any questions or suggestions feel free to reach me !

dobrishinov committed:
UI improvements to be more consistent with the other pages

dobrishinov commented:
Hi @tprouvot,
Thanks for the suggestions - I’ve updated the UI and it looks much more polished now 😊
By the way, the JSON button wasn’t meant for export, it’s just a result view for anyone who needs to inspect the hierarchy in raw JSON.

<img width="1439" height="529" alt="image" src="https://github.com/user-attachments/assets/b071f66f-7db6-4804-9549-7245125e5611" />

dobrishinov committed:
UI changes. Result number in the bottom left using slds-badge

dobrishinov committed:
Small css fix in order to be more consistent

dobrishinov committed:
UI fixes, changed some texts

dobrishinov committed:
UI optimisations, added "Show JSON button"

dobrishinov committed:
Small Css fixes to clean some margins

dobrishinov committed:
Css fixes

tprouvot committed:
Refactor data-load.js to use isRecordId from utils.js and remove unused CSS rules from dependencies-explorer.css

tprouvot committed:
Add syntax highlighting to JSON debug output using Prism library

tprouvot committed:
Extract package.xml generation into a reusable utility function in utils.js

tprouvot committed:
Enhance dependencies explorer to persist metadata type and item in URL parameters

tprouvot committed:
Remove inline style from JSON debug output in dependencies explorer

tprouvot committed:
Apply consistent border radius to JSON debug output in dependencies explorer

tprouvot commented on addon/symbols.svg:
symbols.svg is an extract from [SLDS icons](https://design-system-site-summer-21.herokuapp.com/icons/)

Could you reference only SLDS icons and not new ones ?
ie `apex-class` that you are using seems to be `insert_tag_field `

dobrishinov replied:
@tprouvot Thanks for the commits - I’m currently reviewing and testing them, and everything looks to be working great so far.

Some of the symbols I’m using for metadata are not included in the symbols.svg from the design system which you mentioned. Would it be possible to extract those into a separate file, e.g. custom-symbols.svg? I spent a fair amount of time mapping specific symbols to different metadata types, so I’d prefer to keep the existing metadata-to-icon mapping if possible. 😊

dobrishinov replied:
Hi @tprouvot,
I’ve updated the icon mappings in Dependencies Explorer and adjusted the SVG references. I also reverted symbols.svg to the original version, removing my duplicate icons. 🙏
You can take a look when you have time! 😊

tprouvot reviewed (commented):
Hi @dobrishinov 

I made a few changes, I'll let you check the commits.
I have one remark regarding the symbols.svg, could you take a look ? Thanks

dobrishinov committed:
Update CSS: adjust icon sizes, enhance JSON debug output padding, and modify media query breakpoint.

dobrishinov committed:
Refactor icon mappings in Dependencies Explorer, update SVG references for clarity, and revert symbols.svg to remove custom icons

dobrishinov committed:
Organize metadata types in Dependencies Explorer by categorizing them into Code, UI Components, Pages & Layouts, Data Model, Automation, and Resources & Templates for improved clarity and structure.

dobrishinov committed:
Fix dropdown cache not invalidating on metadata type change.
Added selectedMetadataType to filterDropdownItems memo dependencies.
Previously, switching between metadata types with equal item counts
would show stale results due to unchanged cache key.

dobrishinov committed:
Add link functionality to dependency section titles in Dependencies Explorer. Updated CSS for link styling and added hover effects for better user experience.

dobrishinov committed:
Add unclickable link styles for unavailable Salesforce URLs in Dependencies Explorer. Updated CSS for better visual feedback and modified link creation logic to handle cases where URLs are not available.

dobrishinov committed:
Implement batching support for Tooling API queries in the Dependencies Explorer to prevent HTTP 414 errors. The query is automatically split into batches whenever the IN (...) clause exceeds the allowable length.
Added a new method for fetching records in chunks and updated existing queries to utilize this method for improved performance and reliability.

Small UI Fixes about "Open in Salesforce" link

dobrishinov commented:
Hi @tprouvot, I found that for orgs with a lot of custom fields I got - HTTP 414 URI Too Long
With many custom fields, the code was building a single Tooling API query with a very long IN (...) list (hundreds or thousands of object IDs). That made the request URL too long and the server responded with HTTP 414 URI Too Long. The same pattern could affect other methods that build large IN clauses. I created a new commit for this fix: https://github.com/tprouvot/Salesforce-Inspector-reloaded/pull/964/commits/61dd027f49d0bb3732ee939a0f1a09cc62bf770b

Changes:
- Added batching (TOOLING_QUERY_BATCH_SIZE = 100) for Tooling API queries.
- Implemented _fetchToolingQueryBatched() to split large queries and merge results.
- Updated multiple methods to use batching for object IDs, field IDs, and names.
- This keeps requests within URL limits and prevents failures in orgs with many custom fields/objects.

dobrishinov committed:
Update Dependencies Explorer styles and add tooltip for namespace badges. Changed badge colors for better visibility and added cursor pointer for interactivity. Enhanced namespace display with tooltips in both the badge and tree views for improved user experience.

tprouvot committed:
Use SLDS badges instead of buttons for 'Referenced By' and 'Depends On' filters

tprouvot committed:
Merge branch 'releaseCandidate' into pr/dobrishinov/964

tprouvot commented:
Adding screenshots for doc references

<img width="1437" height="828" alt="image" src="https://github.com/user-attachments/assets/0e3838b6-84f2-42e1-9163-6d789a4d4555" />

tprouvot committed:
feat: Add Dependencies Explorer documentation and update navigation

- Introduced a new documentation file for the Dependencies Explorer, detailing its features, usage instructions, and supported metadata types.
- Updated the mkdocs.yml file to include the new Dependencies Explorer page in the navigation structure.

tprouvot commented:
Merged ! 
Thanks for your work @dobrishinov 👏 

![](https://media1.giphy.com/media/v1.Y2lkPTc5MGI3NjExcWYwbTJ1azYzdWdoOHN4bXY2ZGF0NTgwaXVtamo1aWY5Nm1ndnhsOCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/cFkiFMDg3iFoI/giphy.gif)

dobrishinov commented:
> Adding screenshots for doc references
> 
> <img alt="image" width="1437" height="828" src="https://private-user-images.githubusercontent.com/35368290/547206916-0e3838b6-84f2-42e1-9163-6d789a4d4555.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA2NTg0MDgsIm5iZiI6MTc3MDY1ODEwOCwicGF0aCI6Ii8zNTM2ODI5MC81NDcyMDY5MTYtMGUzODM4YjYtODRmMi00MmUxLTkxNjMtNmQ3ODlhNGQ0NTU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA5VDE3MjgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwZTZkNzZlMGYwZjM4ZGI5YTE3NDM2YTAzMzQ2NmY1ZWZhZjE1ZDYyYjA3ZmQ2YjU4MmRiMTEwMjgzN2Y0ODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hH_ZokyToTbz2gSlwb7a5o4VwXfLB_weCATr9leswcI">

Hey @tprouvot, thanks for the merge!

I noticed one thing regarding the navigation buttons that were changed to tags in this commit:
https://github.com/tprouvot/Salesforce-Inspector-reloaded/pull/964/commits/92840e7614a513a790f8647515ab0e92873df1e0

Would it be possible to revert them to the previous implementation? It felt more consistent with the rest of the UI.

<img width="1439" height="516" alt="image" src="https://github.com/user-attachments/assets/821354ab-6588-4a91-89a9-1b2170de381c" />

Thanks!
